finetune/nn/target_blocks.py

```diff
@@ -398,14 +390,29 @@ def seq_lab_internal(hidden):
                         )
                     )
                     if targets is not None and i != pad_id:
+                        if class_weights is not None:
```

Comment:
Feels like we could factor this out into a separate function but honestly not worth re-running tests to fix.